src/data/finsupp/basic.lean

```diff
@@ -198,8 +198,8 @@ def single (a : α) (b : M) : α →₀ M :=
   { exact ⟨λ H _, h H.symm, λ H, (H rfl).elim⟩ }
 end⟩
 
-lemma single_apply : single a b a' = if a = a' then b else 0 :=
-rfl
+lemma single_apply [decidable (a = a')] : single a b a' = if a = a' then b else 0 :=
```

Comment:
Just to be sure: from `decidable_eq` we can conclude `decidable (a = a')`, right?

Reply:
Yes, `decidable_eq`  is defeq to `Π i j, decidable (i = j)` and reducible.